src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -56,9 +56,32 @@ internal sealed class MethodBodySemanticModel : MemberSemanticModel
         {
             Debug.Assert((object)owner != null);
             Debug.Assert(owner.Kind == SymbolKind.Method);
+            Debug.Assert(owner.MethodKind is not (MethodKind.LocalFunction or MethodKind.AnonymousFunction));
             Debug.Assert(syntax != null);
             Debug.Assert(parentRemappedSymbolsOpt is null || IsSpeculativeSemanticModel);
             Debug.Assert((syntax.Kind() == SyntaxKind.CompilationUnit) == (!IsSpeculativeSemanticModel && owner is SynthesizedSimpleProgramEntryPointSymbol));
+            Debug.Assert(!(IsSpeculativeSemanticModel && owner is SourcePropertyAccessorSymbol { Property.IsIndexer: false }) || hasSpeculativeFieldKeywordBinderInChain(rootBinder));
+
+            static bool hasSpeculativeFieldKeywordBinderInChain(Binder rootBinder)
```

Comment:
>hasSpeculativeFieldKeywordBinderInChain [](http://example.com/codeflow?start=24&length=39)

Is this a duplicate of a helper used in ```AttributeSemanticModel```. As I mentioned earlier, it looks like we could move the assert to the base constructor and share it across all similar semantic models.